app/src/main/java/ai/elimu/appstore/synchronization/AppListArrayAdapter.java

```diff
@@ -362,7 +361,11 @@ protected Integer doInBackground(ApplicationVersion... applicationVersions) {
                         byteArrayOutputStream.write(buffer, 0, bytesRead);
 
                         fileSizeInKbsDownloaded += (bytesRead / 1024);
-                        publishProgress(fileSizeInKbsDownloaded);
+
+                        //Only update APK size every 100KB downloaded
+                        if (fileSizeInKbsDownloaded % 100 == 0) {
```

Comment:
Is this the code change that fixed the laggy UI?

Reply:
Yes.